src/ScriptBuilderTask/ScriptBuilderTask.csproj

```diff
@@ -27,7 +27,7 @@
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.1' ">
     <!--Package needs to stay on 15.x to support Visual Studio 2017 -->
```

Comment:
@bording given that this has `PrivateAssets="All"`, why is this the case?

Reply:
It's true for all the same reasons that analyzers need to be locked to specific versions.

The package versions roughly correspond to MSBuild versions, so you want use the oldest one that corresponds to the minimum version you want to support. In this case it's the MSBuild that would be part of Visual Studio 2017, which is the 15.x packages.

The package references are `PrivateAssets="All"` to avoid having consumers of your package end up with unnecessary MSBuild dependencies in their projects. When the task assembly is loaded into MSBuild, it uses the versions of those assemblies that shipped along with `MSBuild.exe`.

Similar reasoning for the comment further up about specific target frameworks. You need to be using old enough frameworks to work with the oldest version of MSBuild you want to work with. An easy way to figure that out is too look at what assemblies are shipped in the referenced MSBuild packages.

Reply:
@bording thanks, I _think_ get it. It's just a bit difficult to get my head around the fact that we have a _runtime_ package dependency which is not listed as a dependency of package because that dependency is kind of "magically" provided by MSBuild at runtime. I'll go ahead and close this PR, since we don't want this upgrade.